tests/components/samsungtv/test_config_flow.py

```diff
@@ -0,0 +1,388 @@
+"""Tests for Samsung TV config flow."""
+from unittest.mock import call, patch
+
+from asynctest import mock
+import pytest
+from samsungctl.exceptions import AccessDenied, UnhandledResponse
+
+from homeassistant.components.samsungtv.const import (
+    CONF_MANUFACTURER,
+    CONF_MODEL,
+    DOMAIN,
+)
+from homeassistant.components.ssdp import (
+    ATTR_SSDP_LOCATION,
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_MANUFACTURER,
+    ATTR_UPNP_MODEL_NAME,
+    ATTR_UPNP_UDN,
+)
+from homeassistant.const import CONF_HOST, CONF_ID, CONF_METHOD, CONF_NAME
+
+MOCK_USER_DATA = {CONF_HOST: "fake_host", CONF_NAME: "fake_name"}
+MOCK_SSDP_DATA = {
+    ATTR_SSDP_LOCATION: "https://fake_host:12345/test",
+    ATTR_UPNP_FRIENDLY_NAME: "[TV]fake_name",
+    ATTR_UPNP_MANUFACTURER: "fake_manufacturer",
+    ATTR_UPNP_MODEL_NAME: "fake_model",
+    ATTR_UPNP_UDN: "uuid:fake_uuid",
+}
+MOCK_SSDP_DATA_NOPREFIX = {
+    ATTR_SSDP_LOCATION: "http://fake2_host:12345/test",
+    ATTR_UPNP_FRIENDLY_NAME: "fake2_name",
+    ATTR_UPNP_MANUFACTURER: "fake2_manufacturer",
+    ATTR_UPNP_MODEL_NAME: "fake2_model",
+    ATTR_UPNP_UDN: "fake2_uuid",
+}
+
+AUTODETECT_WEBSOCKET = {
+    "name": "HomeAssistant",
+    "description": "HomeAssistant",
+    "id": "ha.component.samsung",
+    "method": "websocket",
+    "port": None,
+    "host": "fake_host",
+    "timeout": 1,
+}
+AUTODETECT_LEGACY = {
+    "name": "HomeAssistant",
+    "description": "HomeAssistant",
+    "id": "ha.component.samsung",
+    "method": "legacy",
+    "port": None,
+    "host": "fake_host",
+    "timeout": 1,
+}
+
+
+@pytest.fixture(name="remote")
+def remote_fixture():
+    """Patch the samsungctl Remote."""
+    with patch("samsungctl.Remote") as remote_class, patch(
+        "homeassistant.components.samsungtv.config_flow.socket"
+    ) as socket_class:
+        remote = mock.Mock()
+        remote.__enter__ = mock.Mock()
+        remote.__exit__ = mock.Mock()
+        remote_class.return_value = remote
+        socket = mock.Mock()
+        socket_class.return_value = socket
+        yield remote
+
+
+async def test_user(hass, remote):
+    """Test starting a flow by user."""
+
+    # show form
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": "user"}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+
+    # entry was added
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input=MOCK_USER_DATA
+    )
+    assert result["type"] == "create_entry"
+    assert result["title"] == "fake_name"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4795320)